src/components/AvatarWithIndicator.js

```diff
@@ -14,37 +19,116 @@ const propTypes = {
     // avatar size
     size: PropTypes.string,
 
+    // Whether we show the sync indicator
+    isSyncing: PropTypes.bool,
 };
 
 const defaultProps = {
     isActive: false,
     size: 'default',
+    isSyncing: false,
 };
 
-const AvatarWithIndicator = ({
-    isActive,
-    source,
-    size,
-}) => {
-    const indicatorStyles = [
-        size === 'large' ? styles.statusIndicatorLarge : styles.statusIndicator,
-        isActive ? styles.statusIndicatorOnline : styles.statusIndicatorOffline,
-    ];
-
-    return (
-        <View
-            style={[size === 'large' ? styles.avatarLarge : styles.sidebarAvatar]}
-        >
-            <Avatar
-                style={[size === 'large' ? styles.avatarLarge : null]}
-                source={source}
-            />
-            <View style={StyleSheet.flatten(indicatorStyles)} />
-        </View>
-    );
-};
+class AvatarWithIndicator extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.rotate = new Animated.Value(0);
+        this.scale = new Animated.Value(1);
+        this.startSyncIndicator = this.startSyncIndicator.bind(this);
+        this.stopSyncIndicator = this.stopSyncIndicator.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.isSyncing) {
+            this.startSyncIndicator();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
```

Comment:
Since this is a `PureComponent`, do you need to implement `shouldComponentUpdate` to make sure it re-renders if `this.props.isSyncing` changes? If it tests fine without it then no need, just thinking out loud.